MIC_U: Stub service funcions

mailwl opened the pull request:
It's just stub to avoid errors, related with not mapped memory and not created event.

after update to SDL 2.0.5 (current release) will be possible to implement audio capturing

wwylele commented on src/core/hle/service/mic_u.cpp:
```diff
@@ -2,37 +2,331 @@
+    ENCODING_PCM8 = 0,
+    ENCODING_PCM16 = 1,
+    ENCODING_PCM8_SIGNED = 2,
+    ENCODING_PCM16_SIGNED = 3
```
Use `CamelCase` for enum members

wwylele commented on src/core/hle/service/mic_u.cpp:
```diff
@@ -2,37 +2,331 @@
+static Encoding encoding;
+static SampleRate sample_rate;
+static s32 offset;
+static u32 size;
```
`offset`, `size` are really not good names for static global variables IMO. They are not descriptive and can be easily hide by local variable, which tends to cause bugs.

wwylele commented on src/core/hle/service/mic_u.cpp:
```diff
@@ -2,37 +2,331 @@
+    if (shared_memory) {
+        shared_memory->name = "MIC_U:shared_memory";
+        shared_memory->Map(Kernel::g_current_process.get(), 0, shared_memory->permissions,
+                           shared_memory->other_permissions);
```
I feel like this is a wrong use of shared memory. My first impression to this service funtion is that, the shared memory is created by application from its memory space, and MIC system module maps the memory to its space using the input shared memory object. If what I think is true, then our HLE service don't need to map the memory at all, but just to operate the memory directly using shared memory interface.

wwylele commented on src/core/hle/service/mic_u.cpp:
```diff
@@ -2,37 +2,331 @@
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    if (shared_memory) {
+        shared_memory->Unmap(Kernel::g_current_process.get(), shared_memory->base_address);
```
ditto

mailwl commented:
addressed the comments.

i remove memory map&unmap, but i think, it's not rights, because Nintendogs don't shows i lot of unmapped read errors without it. But, for Nintendogs games not enough just mic:u service, Y2R completion must be signaled async.

Anyway, as is, PR fixes #2138, with Shader JIT disabled game works.

mailwl commented:
hmm... mic sample from devkitPro don't show any memory errors...

wwylele commented on src/core/hle/service/mic_u.cpp:
```diff
@@ -2,37 +2,320 @@
 
 namespace MIC_U {
 
+enum class Encoding : u8 { PCM8 = 0, PCM16 = 1, PCM8Signed = 2, PCM16Signed = 3 };
```
You can add a trailing comma to let clang-format know you want to expand it
```c++
enum class Encoding : u8 {
    PCM8 = 0,
    PCM16 = 1,
    PCM8Signed = 2,
    PCM16Signed = 3,
};

mailwl replied:
thanks!

Subv commented:
Please fix the build and squash the commits.

mailwl commented:
done

Subv commented:
LGTM

wwylele commented on src/core/hle/service/mic_u.cpp:
```diff
@@ -2,37 +2,325 @@
+
+    cmd_buff[1] = RESULT_SUCCESS.raw; // No error
+    is_sampling = true;
+    LOG_WARNING(Service_MIC, "(STUBBED) called");
```
Log the inputs here?

wwylele commented on src/core/hle/service/mic_u.cpp:
```diff
@@ -2,37 +2,325 @@
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+    sample_rate = static_cast<SampleRate>(cmd_buff[1] & 0xFF);
+    cmd_buff[1] = RESULT_SUCCESS.raw; // No error
+    LOG_WARNING(Service_MIC, "(STUBBED) called");
```
ditto. will stop commenting on same things

mailwl commented:
done
